From d6c404d8b2954fdb90f23d39ca9c436a2f453c68 Mon Sep 17 00:00:00 2001
From: qwang2 <quanyang.wang@windriver.com>
Date: Tue, 26 Jun 2018 17:43:14 +0800
Subject: [PATCH] can: flexcan: add missing FLEXCAN_MCR_IRMQ

There is no deletion of FLEXCAN_MCR_IRMQ in commit 2166556c18ad
("MLK-10131 ENGR00286724-8 can: flexcan: add self wakeup support")
in git://git.freescale.com/imx/linux-imx.git. The FLEXCAN_MCR_IRMQ
is deleted by handling context by mistake. Without setting this bit,
can devices will only receive one packet and stop. So add it back.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/can/flexcan.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/can/flexcan.c b/drivers/net/can/flexcan.c
index d7165dc..daa079e 100644
--- a/drivers/net/can/flexcan.c
+++ b/drivers/net/can/flexcan.c
@@ -964,7 +964,7 @@ static int flexcan_chip_start(struct net_device *dev)
 	reg_mcr = flexcan_read(&regs->mcr);
 	reg_mcr &= ~FLEXCAN_MCR_MAXMB(0xff);
 	reg_mcr |= FLEXCAN_MCR_FRZ | FLEXCAN_MCR_HALT | FLEXCAN_MCR_SUPV |
-		FLEXCAN_MCR_SUPV | FLEXCAN_MCR_WRN_EN |
+		FLEXCAN_MCR_SUPV | FLEXCAN_MCR_WRN_EN | FLEXCAN_MCR_IRMQ |
 		FLEXCAN_MCR_IDAM_C | FLEXCAN_MCR_SRX_DIS |
 		FLEXCAN_MCR_WAK_MSK | FLEXCAN_MCR_SLF_WAK |
 		FLEXCAN_MCR_MAXMB(FLEXCAN_TX_MB_OFF_FIFO);
-- 
1.7.9.5

